File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for(String str : strs){
            // sort str after making it into char array
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String key = new String(chars);

            // if char array exists, add the str to it's respective group
            map.computeIfAbsent(key, k -> new ArrayList<>()).add(str);
        }

        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = {}
        
        for string in strs:
            sorted_string = ''.join(sorted(list(string)))
            
            if sorted_string in anagrams.keys():
                anagrams[sorted_string].append(string)
            else:
                anagrams[sorted_string] = [string]
        
        result = []
        for lists in anagrams.values():
            result.append(lists)
        
        return result